config.go

```diff
@@ -153,6 +153,7 @@ func (c *config) discoverInternalComponents() error {
 	for builder := range command.Builders {
 		builder := builder
 		if !c.Plugins.Builders.Has(builder) {
+			command.TrackBundledPlugin(fmt.Sprintf("packer-builder-%s", builder))
```

Comment:
In testing I only see the warning happening for JSON templates. Looking at this code it looks lie we only track bundled plugins when using command.Builders, which is just for old core right. I think with the telemetry we set this for HCL now but maybe this happens after.

Reply:
I'm surprised this doesn't work on HCL2 templates in your testing, my templates behave the same way in both cases, so I wonder why that is.
Note that this warning only occurs if the plugins are not installed, so if you had installed a version of a plugin with `packer plugins install`, the next `build` that relies on this plugin will effectively not rely on the bundled plugins, and we won't see the message anymore.

Reply:
I'll try again. I might have some plugins laying around